src/main/java/org/benetech/mathshare/converters/Base32TableConverterUtils.java

```diff
@@ -0,0 +1,28 @@
+package org.benetech.mathshare.converters;
+
+import java.util.Locale;
+
+public final class Base32TableConverterUtils {
```

Comment:
lets just use apache commons codec: https://stackoverflow.com/questions/21515479/encode-string-to-base32-string-in-java

Reply:
In our case using Apache common codec is more complex than just converting it by yourself. Not knowing why some char values get duplicated, and for example '0' is converted to 'A', but '1' is being converted to 'C' (not 'B'). I even wanted to use their converter instead of our logic in UrlConverter, but there were similar problems and additionally limit values were not converted properly. The problem also might happen because I had to convert the byte[] to long and reverse. After all, I decided to provide exhaustive tests and implement it myself.

Reply:
Are you using it correctly? you need to pass the long as a byte array

Reply:
I was passing it as a byte[], but if we prefer using this codec I might spend more time on it to figure out what went wrong here.